trussworks/react-uswds
# feat: New Component: Breadcrumb

# Summary

Add [USWDS Breadcrumb Component](https://designsystem.digital.gov/components/breadcrumb/).

## Related Issues or PRs

Fixes #337 

## How To Test

Storybook remains the best way to interact with the component, especially regarding meta data. I would recommend inspecting the elements from storybook to verify:

```
git checkout --track origin/bl-breadcrumb-337
yarn storybook
```

There are also several jest tests that cover basic rendering and uswds class application.

### Screenshots (from storybook)

---

**Default Breadcrumb:**
![image](https://user-images.githubusercontent.com/15805554/105371640-cc132a80-5bd2-11eb-9c1c-5c7b943688c7.png)

Note: non-wrapping breadcrumbs truncate:
![image](https://user-images.githubusercontent.com/15805554/105371839-0250aa00-5bd3-11eb-9ab5-0d4fffb89c00.png)

---

**Breadcrumb with RDFA Metadata:**
![image](https://user-images.githubusercontent.com/15805554/105372032-33c97580-5bd3-11eb-8af2-19a660971ef1.png)
![image](https://user-images.githubusercontent.com/15805554/105372309-725f3000-5bd3-11eb-8b3d-5f7da2a17193.png)

---

**Wrapping Breadcrumb:**
![image](https://user-images.githubusercontent.com/15805554/105372422-9458b280-5bd3-11eb-9f13-5581c504e02c.png)

![image](https://user-images.githubusercontent.com/15805554/105372485-a5092880-5bd3-11eb-87e8-42bcebb5c4a2.png)

---

**Breadcrumb using custom Link components:**
![image](https://user-images.githubusercontent.com/15805554/105372978-2bbe0580-5bd4-11eb-8c9d-5be3b35abc9d.png)

---
